velox/functions/sparksql/String.h

```diff
@@ -283,6 +283,83 @@ struct EndsWithFunction {
   }
 };
 
+/// substring_index function
+/// substring_index(string, string, int) -> string
+/// substring_index(str, delim, count) - Returns the substring from str before
+/// count occurrences of the delimiter delim. If count is positive, everything
+/// to the left of the final delimiter (counting from the left) is returned. If
+/// count is negative, everything to the right of the final delimiter (counting
+/// from the right) is returned. The function substring_index performs a
+/// case-sensitive match when searching for delim.
+template <typename T>
+struct SubstringIndexFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // Results refer to strings in the first argument.
+  static constexpr int32_t reuse_strings_from_arg = 0;
+
+  // ASCII input always produces ASCII result.
+  static constexpr bool is_default_ascii_behavior = true;
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& str,
+      const arg_type<Varchar>& delim,
+      const int32_t& count) {
+    doCall<false>(result, str, delim, count);
+  }
+
+  FOLLY_ALWAYS_INLINE void callAscii(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& str,
+      const arg_type<Varchar>& delim,
+      const int32_t& count) {
+    doCall<true>(result, str, delim, count);
+  }
+
+  template <bool isAscii>
+  FOLLY_ALWAYS_INLINE void doCall(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& str,
+      const arg_type<Varchar>& delim,
+      const int32_t& count) {
+    if (count == 0) {
+      result.setEmpty();
+      return;
+    }
+
+    int64_t index;
+    if (count > 0) {
+      index = stringImpl::stringPosition<isAscii, true>(str, delim, count);
+    } else {
+      index = stringImpl::stringPosition<isAscii, false>(str, delim, -count);
+    }
+
+    auto start = 1;
+    auto length = stringImpl::length<isAscii>(str);
+    auto delimLength = stringImpl::length<isAscii>(delim);
+
+    // Update the start and length based on the count value. And if the index ==
+    // 0, it will return the input str directly and no need to update the start
+    // and length.
+    if (index != 0) {
+      if (count > 0) {
+        length = index - 1;
+      } else {
+        start = index + delimLength;
+        length = length - index - delimLength + 1;
+      }
+    }
+
+    auto byteRange =
+        stringCore::getByteRange<isAscii>(str.data(), start, length);
+
+    // Generating output string.
```

Comment:
This comment is redundant. Let's remove.

Reply:
Removed.